parlai/core/agents.py

```diff
@@ -113,8 +136,9 @@ def shutdown(self):
 
 class Teacher(Agent):
     """
-    Basic Teacher agent which keeps track of how many times it's received
-    messages. Teachers provide the ``report()`` method to get back metrics.
+    Basic Teacher agent that keeps track of how many times it's received messages.
+
```

Comment:
why does flake8 put these giant spaces between sentences?

Reply:
https://www.python.org/dev/peps/pep-0257/

Docstrings should be a one-line summary, possibly followed by a longer description or additional details.